From: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Date: Sun, 18 Jan 2026 15:24:46 +0100
Subject: [PATCH] serial: 8250: add UPIO_AU case to set_io_from_upio()

This patch fixes the "Unsupported UART type 4" error by adding support
for the UPIO_AU I/O type in set_io_from_upio() (used by Au1x00 and
RT288x). It assigns the correct serial input/output functions
(io_serial_in, io_serial_out) for proper operation when UPIO_AU is
configured.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
---
 drivers/tty/serial/8250/8250_port.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -453,6 +453,7 @@ static void set_io_from_upio(struct uart
 		break;
 #ifdef CONFIG_HAS_IOPORT
 	case UPIO_PORT:
+	case UPIO_AU:
 		p->serial_in = io_serial_in;
 		p->serial_out = io_serial_out;
 		break;
